shared/Microsoft.AspNetCore.RangeHelper.Sources/RangeHelper.cs

```diff
@@ -21,15 +22,23 @@ internal static class RangeHelper
         /// </summary>
         /// <param name="context">The <see cref="HttpContext"/> associated with the request.</param>
         /// <param name="requestHeaders">The <see cref="RequestHeaders"/> associated with the given <paramref name="context"/>.</param>
+        /// <param name="length">The total length of the file representation requested.</param>
         /// <param name="lastModified">The <see cref="DateTimeOffset"/> representation of the last modified date of the file.</param>
         /// <param name="etag">The <see cref="EntityTagHeaderValue"/> provided in the <see cref="HttpContext.Request"/>.</param>
-        /// <returns>A collection of <see cref="RangeItemHeaderValue"/> containing the ranges parsed from the <paramref name="requestHeaders"/>.</returns>
-        public static ICollection<RangeItemHeaderValue> ParseRange(HttpContext context, RequestHeaders requestHeaders, DateTimeOffset? lastModified = null, EntityTagHeaderValue etag = null)
+        /// <returns>A boolean value which represents if the <paramref name="requestHeaders"/> contain a single valid 
+        /// range request (true), or not (false). A <see cref="RangeItemHeaderValue"/> which represents the normalized form of the 
```

Comment:
This sounds verbose/wordy. Any ideas for a better design than returning a tuple? Or any suggestions for the doc comment? @Tratcher 
```
        /// <summary>
        /// Returns the normalized form of the requested range if the Range Header in the <see cref="HttpContext.Request"/> is valid.
        /// </summary>
        /// <param name="context">The <see cref="HttpContext"/> associated with the request.</param>
        /// <param name="requestHeaders">The <see cref="RequestHeaders"/> associated with the given <paramref name="context"/>.</param>
        /// <param name="length">The total length of the file representation requested.</param>
        /// <param name="lastModified">The <see cref="DateTimeOffset"/> representation of the last modified date of the file.</param>
        /// <param name="etag">The <see cref="EntityTagHeaderValue"/> provided in the <see cref="HttpContext.Request"/>.</param>
        /// <returns>A boolean value which represents if the <paramref name="requestHeaders"/> contain a single valid 
        /// range request (<c>true</c>), or not (<c>false</c>). A <see cref="RangeItemHeaderValue"/> which represents the normalized form of the 
        /// range parsed from the <paramref name="requestHeaders"/>.</returns>
        /// <remark>If the Range header is not null or empty but the range cannot be parsed correctly, or if the provided length is 0, the range request cannot be satisfied. 
        /// The method returns (<c>true</c>,<c>null</c>) resulting in <see cref="StatusCodes.Status416RangeNotSatisfiable"/>.</remark>
```